doc/tutorial/read_mapper/read_mapper_step1.cpp

```diff
@@ -0,0 +1,59 @@
+#include <seqan3/argument_parser/all.hpp>
+#include <seqan3/io/stream/debug_stream.hpp>
+
+using namespace seqan3;
+
+void run_program(std::filesystem::path const & reference_path,
+                 std::filesystem::path const & query_path,
+                 std::filesystem::path const & sam_path,
+                 uint8_t errors)
+{
+    debug_stream << "reference_file_path: " << reference_path << '\n';
+    debug_stream << "query_file_path:     " << query_path << '\n';
+    debug_stream << "sam_file_path        " << sam_path << '\n';
+    debug_stream << "errors:              " << errors << '\n';
+}
+
+struct cmd_arguments
+{
+    std::filesystem::path reference_path{};
+    std::filesystem::path query_path{};
+    std::filesystem::path sam_path{"out.sam"};
+    uint8_t errors{0};
+};
+
+void initialise_argument_parser(argument_parser & parser, cmd_arguments & args)
+{
+    parser.info.author = "E. coli";
+    parser.info.short_description = "Map reads against a reference.";
+    parser.info.version = "1.0.0";
+    parser.add_option(args.reference_path, 'r', "reference", "The path to the reference.", option_spec::REQUIRED,
+                      file_ext_validator({"fa","fasta"}) | seqan3::file_existance_validator());
```

Comment:
```suggestion
                      file_ext_validator{{"fa","fasta"}} | seqan3::file_existance_validator{});
```
and at all other places

Reply:
Works for the existence validator.

`file_ext_validator{{"fa","fasta"}}` will give you
```
terminate called after throwing an instance of 'std::length_error'
  what():  basic_string::_M_create
Abort trap: 6
```